### PR TITLE
Add jest unit testing to Chip component

### LINKED ISSUE #3557: Chip: Add test specifications
### Describe the feature you would like to see added

Add Chip component unit testing with Jest

### Is your feature request related to a problem?

_No response_

### Describe the solution you'd like

_No response_

### Describe alternatives you have considered

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by habubey**
### Feature Requests
Fix #3557 Add jest unit test to Chip component

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -20,6 +20,7 @@
         "test:check": "jest --watchAll=false --coverage"
     },
     "dependencies": {
+        "@testing-library/user-event": "^14.4.3",
```
Comment (melloware) [inline package.json:23]: This should go in Dev dependencies.

File: components/lib/chip/Chip.spec.js [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+        expect(wrapper.length).toBe(1);
+    });
+
+    test('when label is property it returns with label class', () => {
+        const { container } = render(<Chip label={'test'} />);
+
+        const label = container.getElementsByClassName('p-chip-text');
+
+        expect(label.length).toBe(1);
```
Comment (melloware) [inline components/lib/chip/Chip.spec.js:40]: You should check the label text is `'test'` not just that is has a label

File: components/lib/chip/Chip.spec.js [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+
+        expect(chipImage.getAttribute('alt')).toBe('chip');
+    });
+
+    test('when icon is property it returns with icon class', () => {
+        const { container } = render(<Chip icon={'pi pi-check'} />);
+
+        const wrapper = container.getElementsByClassName('p-chip-icon');
+
+        expect(wrapper.length).toBe(1);
```
Comment (melloware) [inline components/lib/chip/Chip.spec.js:32]: Check the icon value is correct

**Review by melloware — CHANGES_REQUESTED** [review]

File: components/lib/chip/Chip.spec.js [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+describe('Chip', () => {
+    test('when removable is true it returns with remove icon', () => {
+        const removeOn = jest.fn();
+        const { container } = render(<Chip removable onRemove={removeOn} />);
+        const chipRemoveIcon = container.getElementsByClassName('p-chip-remove-icon')[0];
+
+        fireEvent.click(chipRemoveIcon);
+
+        expect(container.getElementsByClassName('p-chip-remove-icon pi pi-times-circle').length).toBe(0);
+        expect(removeOn).toHaveBeenCalledTimes(1);
```
Comment (melloware) [inline components/lib/chip/Chip.spec.js:15]: We can decide as a team or @mertsincan can decide but I am big fan of "arrange act assert" so something like this.

```js
        // Arrange
        const removeOn = jest.fn();
        const { container } = render(<Chip removable onRemove={removeOn} />);
        const chipRemoveIcon = container.getElementsByClassName('p-chip-remove-icon')[0];

        // Act
        fireEvent.click(chipRemoveIcon);

        // Assert
        expect(container.getElementsByClassName('p-chip-remove-icon pi pi-times-circle').length).toBe(0);
        expect(removeOn).toHaveBeenCalledTimes(1);
```

File: components/lib/chip/Chip.spec.js [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+    });
+
+    test('when image is property it returns with image class', () => {
+        const { container } = render(<Chip image={'test'} />);
+
+        const wrapper = container.getElementsByClassName('p-chip p-component p-chip-image');
+        const chipImage = within(wrapper[0]).getByRole('img');
+
+        expect(chipImage.getAttribute('alt')).toBe('chip');
+    });
```
Comment (melloware) [inline components/lib/chip/Chip.spec.js:25]: We should add a test for the `imageAlt` property of the Chip as well.

**Comment by melloware** [comment]
I am going to merge this and then update some changes

File: components/lib/chip/Chip.spec.js (hunk 4 above)
Comment (bahadirsofuoglu) [reply components/lib/chip/Chip.spec.js:15]: Hi, Testing library is a legit library. Buğra makes this understandable by using the right spaces. Reviews provide that too. With all due respect, I don't believe this is necessary. I wonder why you think it's necessary.

Comment (melloware) [reply components/lib/chip/Chip.spec.js:15]: @bahadirsofuoglu it is a very common pattern in unit testing in C#/Java etc.

Here is a great explanation: https://docs.telerik.com/devtools/justmock/basic-usage/arrange-act-assert#:~:text=Benefits%20of%20Using%20Arrange%20Act%20Assert&text=Clarifies%20and%20focuses%20attention%20on,many%20different%20things%20at%20once.

**Pull request merged**